File: backend/src/domain/entities/MentorExpertise.ts
import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm';

@Entity('mentor_expertise')
export class MentorExpertise {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'mentor_member_id' })
  mentorMemberId: number;

  @Column({ type: 'varchar', nullable: true })
  topic: string;

  @Column({ type: 'integer', name: 'years_experience', nullable: true })
  yearsExperience: number;

  @Column({ type: 'text', nullable: true })
  description: string;
}

File: backend/src/main/java/com/service/backend/shared/entity/MentorExpertise.java
package com.service.backend.shared.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("mentor_expertise")
public class MentorExpertise {

    @Id
    private Integer id;

    @Column("mentor_member_id")
    private Integer mentorMemberId;

    private String topic;

    @Column("years_experience")
    private Integer yearsExperience;

    private String description;

    private String category;

    private String tag;
}
